tests_e2e/orchestrator/lib/agent_test_suite.py

```
@@ -259,17 +292,16 @@ def execute_script_on_node(self, script_path: Path, parameters: str = "", sudo:
 
         result = custom_script.run(parameters=parameters, sudo=sudo)
 
-        if result.exit_code != 0:
-            output = result.stdout if result.stderr == "" else f"{result.stdout}\n{result.stderr}"
-            raise Exception(f"[{command_line}] failed:\n{output}")
-
         if result.stdout != "":
             separator = "\n" if "\n" in result.stdout else " "
             log.info("stdout:%s%s", separator, result.stdout)
         if result.stderr != "":
             separator = "\n" if "\n" in result.stderr else " "
             log.error("stderr:%s%s", separator, result.stderr)
 
+        if result.exit_code != 0:
+            raise Exception(f"[{command_line}] failed. Exit code: {result.exit_code}")
+
```

We need to raise the exception *after* logging the command's output